plugins/docker_binary_builder/app/models/binary_builder.rb

```diff
@@ -0,0 +1,136 @@
+class BinaryBuilder
+  DOCKER_BUILD_FILE = 'Dockerfile.build'.freeze
+  BUILD_SCRIPT = '/app/build.sh'.freeze
+  ARTIFACTS_FILE = 'artifacts.tar'.freeze
+  ARTIFACTS_FILE_PATH = "/app/#{ARTIFACTS_FILE}".freeze
+  DOCKER_HOST_CACHE_DIR = '/opt/samson_build_cache'.freeze
```

Comment:
Ok, so the main idea of this was to allow projects to share the .m2, .ivy to the projects. If we would expose these as ENV variables like M2_HOME or IVY_HOME when we start the container, the tools will automatically pickup where the cache is, and developer won't even need to know that they are being cached. This will be completely transparent to him.

As a big advantage we can think that a lot of JVM apps share lots of common dependencies (logging, etc..) so less space used than one directory per project, less probability of running out of space in the DockerHost machine.

As a disadvantage we can't protect against what @jonmoter said, i.e, a bad intentioned developer. As with this cache approach everything should be completely transparent to the developer, only in an intentioned way a developer could do it. So the problem still exists but with a lot smaller probability.

Creating a project specific directory with cached dependencies will handle the "security" problem, sure, but the disk space might become an issue.